packages/builder/src/pages/builder/app/[application]/settings/oauth2/OAuth2ConfigModalContent.svelte

```diff
@@ -0,0 +1,144 @@
+<script lang="ts">
+  import { oauth2 } from "@/stores/builder"
+  import type { OAuth2Config, UpsertOAuth2Config } from "@/types"
+  import {
+    Body,
+    Divider,
+    Heading,
+    Input,
+    keepOpen,
+    Link,
+    ModalContent,
+    notifications,
+  } from "@budibase/bbui"
+  import { PASSWORD_REPLACEMENT } from "@budibase/types"
+  import type { ZodType } from "zod"
+  import { z } from "zod"
+
+  export let config: OAuth2Config | undefined = undefined
+
+  let errors: Record<string, string> = {}
+  let hasBeenSubmitted = false
+
+  $: data = (config as Partial<OAuth2Config>) ?? {}
+
+  $: isCreation = !config
+  $: title = isCreation
+    ? "Create new OAuth2 connection"
+    : "Edit OAuth2 connection"
+
+  const requiredString = (errorMessage: string) =>
+    z.string({ required_error: errorMessage }).trim().min(1, errorMessage)
+
+  const validateConfig = (config: Partial<OAuth2Config>) => {
+    const validator = z.object({
+      name: requiredString("Name is required.").refine(
+        val =>
+          !$oauth2.configs
+            .filter(c => c.id !== config.id)
+            .map(c => c.name.toLowerCase())
+            .includes(val.toLowerCase()),
+        {
+          message: "This name is already taken.",
+        }
+      ),
+      url: requiredString("Url is required.").url(),
+      clientId: requiredString("Client ID is required."),
+      clientSecret: requiredString("Client secret is required."),
+    }) satisfies ZodType<UpsertOAuth2Config>
+
+    const validationResult = validator.safeParse(config)
+    errors = {}
+    if (!validationResult.success) {
+      errors = Object.entries(
+        validationResult.error.formErrors.fieldErrors
+      ).reduce<Record<string, string>>((acc, [field, errors]) => {
+        if (errors[0]) {
+          acc[field] = errors[0]
+        }
+        return acc
+      }, {})
+    }
+
+    return validationResult
+  }
+
+  $: saveOAuth2Config = async () => {
+    hasBeenSubmitted = true
+    const validationResult = validateConfig(data)
+    if (validationResult.error) {
+      return keepOpen
+    }
+
+    try {
+      if (isCreation) {
+        await oauth2.create(validationResult.data)
+        notifications.success("Settings created.")
+      } else {
+        await oauth2.edit(config!.id, validationResult.data)
+        notifications.success("Settings saved.")
+      }
+    } catch (e: any) {
+      notifications.error(e.message)
```

Comment:
Worth adding a user readable message to this as well, like `Failed to save config - ${e.message}`